source/Calamari.Testing/EnvironmentVariables.cs

```diff
@@ -103,12 +125,24 @@ public static string Get(ExternalVariable property)
             }
 
             var valueFromEnv = Environment.GetEnvironmentVariable(attr.Name);
-            if (valueFromEnv == null)
+            if (valueFromEnv != null)
             {
-                throw new Exception($"Environment Variable `{attr.Name}` could not be found. The value can be found in the password store under `{attr.LastPassName}`");
+                return valueFromEnv;
+                //throw new Exception($"Environment Variable `{attr.Name}` could not be found. The value can be found in the password store under `{attr.LastPassName}`");
             }
 
-            return valueFromEnv;
+            if (SecretManagerIsEnabled)
+            {
+                var valueFromSecretManager = string.IsNullOrEmpty(attr.SecretReference)
+                    ? null
+                    : await SecretManagerClient.Value.GetSecret(attr.SecretReference, cancellationToken, throwOnNotFound: false);
+                if (!string.IsNullOrEmpty(valueFromSecretManager))
+                {
+                    return valueFromSecretManager;
+                }
+            }
+
+            throw new Exception($"Unable to find `{attr.Name}` as either an Environment Variable, or a SecretReference. The value can be found in the password store under `{attr.LastPassName}`");
```

Comment:
This may be slightly misleading if `SecretManageIsEnabled` is false? I think there is value in composing this message to say exactly what's happening, even if it only saves a single engineer a moment of debugging pain.

Reply:
Also, consider changing `LastPassName` to something else?